### PR TITLE
Actually implement UnsafeLegacyServerConnect as documented

### LINKED ISSUE #18295: UnsafeLegacyServerConnect option described in SSL_CONF_cmd man page has no effect?
Affected version: 3.0.2

I don't really know if this is a documentation bug or a missing line in `ssl/ssl_conf.c`, but the `UnsafeLegacyServerConnect` option in the configuration does not seem to enable  `SSL_OP_LEGACY_SERVER_CONNECT`

`SSL_OP_LEGACY_SERVER_CONNECT` was set by default before OpenSSL 3.0.0, but now it is off.

I have to connect to an unpatched server with `curl`, and the connection failed with an SSL error:
`error:0A000152:SSL routines::unsafe legacy renegotiation disabled`

I tried to create a custom OpenSSL configuration file as a workaround, but setting the `UnsafeLegacyServerConnect` had no effect. As a temporary solution, I used the `UnsafeLegacyRenegotiation` option, which enabled me to connect to this server.

Will `UnsafeLegacyServerConnect` be supported in an upcoming version, or the man page will be updated? Or am I missing something else?

### PR DESCRIPTION
**Pull request by t8m**
We documented the option in 3.0 but we forgot to implement it.

Fixes #18295

### CODE REVIEW

**Comment by mattcaswell** [comment]
The `-legacy_server_connect`

````
    SSL_CONF_CMD_SWITCH("legacy_server_connect", SSL_CONF_FLAG_CLIENT),
````

and `-no_legacy_server_connect`

````
    SSL_CONF_CMD_SWITCH("no_legacy_server_connect", SSL_CONF_FLAG_SERVER),
````

command line options,  defined in the same file, also look suspect to me. At least one is defined as a client flag and one is defined as a server flag??

File: ssl/ssl_conf.c [hunk 1]
```diff
@@ -384,6 +384,8 @@ static int cmd_Options(SSL_CONF_CTX *cctx, const char *value)
         SSL_FLAG_TBL_SRV("ECDHSingle", SSL_OP_SINGLE_ECDH_USE),
         SSL_FLAG_TBL("UnsafeLegacyRenegotiation",
                      SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION),
+        SSL_FLAG_TBL("UnsafeLegacyServerConnect",
+                     SSL_OP_LEGACY_SERVER_CONNECT),
```
Comment (mattcaswell) [inline ssl/ssl_conf.c:388]: Can we test in some way that this has had an effect?

Comment (t8m) [reply ssl/ssl_conf.c:388]: We can at least test that the SSL_OP is set.

Comment (t8m) [reply ssl/ssl_conf.c:388]: Added a test_ssl_new testcase.

**Comment by t8m** [comment]
I've added a fix for the client/server mismatch for no_legacy_server_connect option.

**Review by kaduk — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for your reviews.

**Pull request closed without merging**